mmdet/models/detectors/deformable_detr.py

```diff
@@ -1,12 +1,598 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+import math
+from typing import Dict, List, Tuple, Union
+
+import torch
+import torch.nn.functional as F
+from mmcv.cnn.bricks.transformer import (FFN, MultiheadAttention,
+                                         MultiScaleDeformableAttention,
+                                         build_norm_layer)
+from mmengine.model import ModuleList, xavier_init
+from torch import Tensor, nn
+from torch.nn.init import normal_
+
 from mmdet.registry import MODELS
-from .detr import DETR
+from mmdet.structures import OptSampleList, SampleList
+from mmdet.utils import OptConfigType
+from ..layers import (DetrTransformerDecoder, DetrTransformerDecoderLayer,
+                      DetrTransformerEncoder, DetrTransformerEncoderLayer,
+                      SinePositionalEncoding, inverse_sigmoid)
+from .base_detr import TransformerDetector
 
 
 @MODELS.register_module()
-class DeformableDETR(DETR):
+class DeformableDETR(TransformerDetector):
     r"""Implementation of `Deformable DETR: Deformable Transformers for
     End-to-End Object Detection <https://arxiv.org/abs/2010.04159>`_"""
 
-    def __init__(self, *args, **kwargs) -> None:
-        super(DETR, self).__init__(*args, **kwargs)
+    def __init__(self,
+                 *args,
+                 decoder_cfg: OptConfigType = None,
+                 bbox_head: OptConfigType = None,
+                 with_box_refine: bool = False,
+                 as_two_stage: bool = False,
+                 num_feature_levels: int = 4,
+                 **kwargs) -> None:
+        self.with_box_refine = with_box_refine
+        self.as_two_stage = as_two_stage
+        self.num_feature_levels = num_feature_levels
+
+        if "with_box_refine" in bbox_head:
```

Comment:
We can only set these three args in the detector level in all our configs and automatically pass them to bboxhead  here, Please modify all related configs